### PR TITLE
Added YML as optional output format

### PR DESCRIPTION
**Pull request by bencolon**
Warning: this PR contains my two previous PRs (bug in running tests & bug in command options)

### CODE REVIEW

File: lib/gemsurance/yml_formatter.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module Gemsurance
+  class YmlFormatter
+    def initialize(gem_infos)
+      @gem_infos = gem_infos
+    end
+
+    def format
+      ERB.new(File.read(File.join(File.dirname(File.expand_path(__FILE__)), 'templates/output.yml.erb')), nil, '-').result(binding)
```
Comment (jonkessler) [inline lib/gemsurance/yml_formatter.rb]: Could you extract a variable or two here? This is super hard to read.

File: lib/gemsurance/templates/output.yml.erb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<%- @gem_infos.sort { |a, b| a.name.downcase <=> b.name.downcase }.each do |gem_info| -%>
```
Comment (jonkessler) [inline lib/gemsurance/templates/output.yml.erb]: I would prefer this happen in yml_formatter.rb.

File: lib/gemsurance/templates/output.yml.erb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+<%= gem_info.name %>:
+  in_gem_file: <%= gem_info.in_gem_file %>
+  bundle_version: <%= gem_info.current_version %>
+  newest_version: <%= gem_info.newest_version %>
+  <%- if gem_info.vulnerable? -%>
```
Comment (jonkessler) [inline lib/gemsurance/templates/output.yml.erb]: Are these hyphens necessary? Could you remove them if not?

**Comment by jonkessler** [comment]
I already merged the options fix, and I"m not sure why the test-unit change is required.. could you remove those? Otherwise, this looks good. Thanks for the contribution.

File: lib/gemsurance/yml_formatter.rb (hunk 1 above)
Comment (bencolon) [inline lib/gemsurance/yml_formatter.rb]: I've just copied current `html_formatter.rb` code, but you're totally right, I'll create a cleaner `BaseFormatter`class and use it in `HtmlFormatter` and `YmlFormatter`

File: lib/gemsurance/templates/output.yml.erb (hunk 2 above)
Comment (bencolon) [inline lib/gemsurance/templates/output.yml.erb]: :+1: Also need to move that in the future `BaseFormatter`class.

File: lib/gemsurance/templates/output.yml.erb (hunk 3 above)
Comment (bencolon) [inline lib/gemsurance/templates/output.yml.erb]: Yes necessary otherwise there are white lines in YML output file.

**Comment by bencolon** [comment]
I will update this PR with your comments ...

**Comment by bencolon** [comment]
Finally, I'll open another PR. Closing this one.

**Pull request closed without merging**